### ISSUE TITLE
Add a wind speed indicator to hourly chart

### ISSUE THREAD

**Issue by 0-5-0** [Akylas/oss-weather#268]
Would it be possible to add a wind speed indicator to the hourly chart? It's nice to see compared to the normal hourly chart the wind gusts or wind speeds.

If you go to Menu -> Chart you can see the wind speeds there but it's not shown in the other views. Not sure if just an icon color change would work for wind speeds. I guess adding a second line for the wind speed would make the scale too large and chart too high otherwise.

Could the scale shown in Menu -> Chart be shown in hourly view for precipitation?

**Comment by farfromrefug** [comment]
@0-5-0 i am ok adding it but i could not find a nice display. If you have an example i can look at it

**Comment by 0-5-0** [comment]
Yes I think it might be difficult to add the actual measurement, would it be possible just to have the wind direction arrow go orange when it's "stronger" winds? Sort of like you do for the weather data wind speed/wind gust icons.

I do personally prefer the hourly graph instead of the hourly chart but I wanted to test the stacked bars :D

**Comment by 0-5-0** [comment]
or it would be adding a second row when there's wind gusts with the same coloring as in the hourly graph view, you already sort of do this in the hourly graph view where some time slots will show both a normal wind and wind gust icon.

**Comment by farfromrefug** [comment]
@0-5-0 the issue with hourly chart is that each "data" has a lot less horizontal space. not enough space to write the speed text

**Comment by 0-5-0** [comment]
Yes, I think just colouring them would give a nice indicator at least. White (0-29km), Yellow/Orange (30-49km), Red (50+km)

**Comment by farfromrefug** [comment]
@0-5-0 good idea will run some tests

**Comment by 0-5-0** [comment]
If you can fit at the same time a second row of arrows for Wind gusts similar to how you do it in the hourly graph but without the speed text then it would be useful too and it shouldn't be too busy since it would only show up below the corresponding Wind speed read out from time to time. Unless you live in a place that has constant wind gusts I guess..

**Comment by 0-5-0** [comment]
I'm not sure if it would be too much colour but in regards to the new colours for the hourly chart wind icons, maybe they could be made to match in the pop up text, I used "Gabriel de Castilla Spanish Antartic Base" as a location with ECMWF 0.25, it says 50km/hr+ which makes the icon orange/yellow, but when you tap the hourly chart the pop up text for wind is white and the text for wind in daily chart below that is also white for that same wind speed.

<img src="https://github.com/user-attachments/assets/5ed3cf17-fd31-40b6-8afb-29ce2434c7e0" width=45% height=45%>

I also used the Brave browser AI chat which spit out these wind speed recommendations, I'm not sure if you have any preference when it comes to those providers :), or what might be a good range if you went with white -> yellow -> orange -> red.

The NWS defines:
Low risk: 0-39 km/h (0-24 mph)
Moderate risk: 40-62 km/h (25-39 mph)
High risk: 63-92 km/h (40-57 mph)
Extreme risk: 93 km/h and above (58 mph and above)

The WMO defines:
Low risk: 0-30 km/h (0-19 mph)
Moderate risk: 31-50 km/h (20-31 mph)
High risk: 51-75 km/h (32-47 mph)
Extreme risk: 76 km/h and above (48 mph and above)

Then there's also the Beaufort wind scale:

0: Calm, less than 1 km/h
1: Light air, 1-5 km/h
2: Light breeze, 6-11 km/h
3: Gentle breeze, 12-19 km/h
4: Moderate breeze, 20-28 km/h
5: Fresh breeze, 29-38 km/h
6: Strong breeze, 39-49 km/h
7: Near gale, 50-61 km/h
8: Gale, 62-74 km/h
9: Strong gale, 75-88 km/h
10: Storm, 89-102 km/h
11: Violent storm, 103-117 km/h
12: Hurricane, 118 km/h and above

**Comment by 0-5-0** [comment]
I still probably prefer the normal hourly "graph" view (not the chart one), since it easily shows the wind speed and wind gusts, but the split precipitation for rain/snow in the hourly "chart" is nice, I'm not sure if the split precipitation could be added to the normal hourly graph (not chart).

**Comment by farfromrefug** [comment]
@0-5-0 i am not sure about wanting the wind speed line to have colors. Wondering if would make "too many" colors.
Will try to run some tests

**Comment by 0-5-0** [comment]
Not necessarily the line but I guess where ever there's a wind speed measurement, to have the icon and text turn to the same color in all places where a wind measurements are shown (hourly chart arrow, hourly chart pop-up, daily list data points).

Where you see "Cloudy" in the pop-up above the hourly chart the arrow and "53km/hr" should probably match the color of the wind arrow in the hourly chart. Then in the daily view for Friday (56km), Saturday (54km), Monday (52km), then the icon and text should match the same color as the others?

Alternatively I'm not sure if only the "Arrow" icon would change color and the text would stay white but when looking at the AQI and precipitation data points both the icons and the text change color for everything else.

**Comment by 0-5-0** [comment]
Looking at it again, I understand also as a design choice it might be fine having the icon and text white everywhere else since the actual wind speed is written, so having the color change only where an icon and no text is written would make sense too.

I'm seeing white icons and white text for all the wind speed measurements, then yellow/orange (whatever that color is) for wind GUSTS, maybe that's the difference I'm seeing?

I thought I had seen red in the past depending on the wind speeds/gusts so I thought it was a ladder of sorts based on the wind speeds.

**Comment by 0-5-0** [comment]
Would the yellow/orange icon in the hourly chart then indicate wind GUSTS are present? Then I might expect the pop-up when you select the specific hour to show both the wind speed and wind gust speeds. It could be a way to display wind gusts in that chart now that you also have the line showing wind speeds.. if that makes any sense :D

**Comment by 0-5-0** [comment]
Ok, so I guess as long as its consistent across everything then that would probably be fine.

Where the wind icon in the 3rd screenshot is yellow in the hourly chart is when I wasn't clear if it was because of the wind speed or wind gust since when you tap the hourly view to get the pop-up there's only 40km/hr wind displayed there in white text where I might have expected to see a yellow wind measurement (speed or gust).

I think since you have the wind line now for the chart view then that icon could probably stay white so that it's consistent with the coloring for wind speed everywhere else. I'm not sure if you could instead print the scale in this hourly chart view, so we could see on the left/right the measurements (like when you press the main menu button -> chart).

Alternatively I'm not sure if the wind speed measurement could be attached to the wind speed line like you do for the temperature measurement. It "could" be a toggle I guess if someone didn't like it.

Maybe also consider adding a third yellow line for wind gusts with the same type of speed printing as you do for the temperature measurement?

To help clean up the chart if you do add more lines I'm not sure if the wind direction could be tagged to the wind speed line as well, like you do for the temperature values, it would remove one full line at the top of the chart and there could only be a new icon where the wind direction changes (if that's possible), so potentially only 3-6 wind icons attached to the line where currently you have one every hour at the top.

When lines overlap though I'm not sure if it would become too busy.

<img src="https://github.com/user-attachments/assets/ea686edc-ee49-40f8-80f7-b4226461cff7" width=33% height=33%>
<img src="https://github.com/user-attachments/assets/3f86f773-4258-4234-adbd-8a31867d076a" width=33% height=33%>
<img src="https://github.com/user-attachments/assets/6c6e8dc1-58bd-4353-b192-7feeac3c53f6" width=33% height=33%>

**Comment by 0-5-0** [comment]
Should the wind speed line also be shown when you open the daily view? I can see it on the main page using the hourly chart, if I do menu -> chart, but it would probably be good to have it also in the daily view's hourly chart.

**Comment by 0-5-0** [comment]
I think this can be closed, Wind speed indicator works well.

The only things I was thinking is to maybe add a wind **gust** indicator to the hourly view "chart" and then the stacked precipitation bars to the normal hourly view "graph", those are the only things missing from each one of the hourly views if you compare them both.

Suggestions for hourly "chart" might be adding wind speed direction to key points on the white line with the wind speed, then maybe the directional arrows near the top could be converted to show only when there's wind gusts, but the icon might have to change to the wind gust icon to avoid confusion. Could be revisited another time or in another discussion :).

I'm liking the hourly graph more and more now.
